subsys/bluetooth/host/gatt.c

```diff
@@ -3083,24 +3110,25 @@ int bt_gatt_subscribe(struct bt_conn *conn,
 		}
 	}
 
-	/* Skip write if already subscribed */
-	if (!has_subscription) {
-		int err;
-
-		err = gatt_write_ccc(conn, params->ccc_handle, params->value,
-				     gatt_write_ccc_rsp, params);
-		if (err) {
-			return err;
-		}
-	}
+	ret = 0;
 
 	/*
 	 * Add subscription before write complete as some implementation were
 	 * reported to send notification before reply to CCC write.
 	 */
 	gatt_subscription_add(conn, params);
 
-	return 0;
+	/* Skip write if already subscribed */
```

Comment:
Why did you move this below `gatt_subscription_add()`?

Reply:
It was because of the paramaters are copied otherwise the response will not be handled properly as it did not have the node entry set.